### PR TITLE
add support for windows terminal hide

### LINKED ISSUE #4: Support Windows Terminal
If the default terminal application is not set to windows console host but to windows terminal, the node application will start inside windows terminal which results in a secondary empty window which can be toggled hidden, but it does not have any effect on the real console window which is hosted inside windows terminal.

<details>
<summary>with windows terminal as default</summary>

https://user-images.githubusercontent.com/50082450/171955404-714519dc-df70-46d6-b4de-bb98c35a2b3c.mp4

</details>

<details>
<summary>with windows console host as default</summary>

https://user-images.githubusercontent.com/50082450/171955430-c4ac9e12-4d71-4e8a-acc3-33f9567b4cef.mp4

</details>

### PR DESCRIPTION
**Pull request by titushm**
fixes #4

### CODE REVIEW

**Comment by titushm** [comment]
Think i have found an alternate method, testing now but seems to work as intended
Tested with:

- Command Prompt ✔
- Command Prompt Admin ✔
- Powershell ✔
- Powershell Admin ✔
- Windows Terminal Command Prompt✔
- Windows Terminal Command Prompt Admin✔
- Windows Terminal Powershell✔
- Windows Terminal Powershell Admin✔

**Comment by hetrodoo** [comment]
Thanks for the PR @titushm, but I have a question regarding your code.

Inside the method `HideConsole`, at the lines 18 and 19, you created a variable `DWORD processId` that is setup shortly after but not referenced anywhere, is there a need for those lines to exist?

```c++
DWORD processId;
GetWindowThreadProcessId(GetTargetWindow(), &processId);
```

File: node-hide-console-window.cc [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+        return consoleWindow;
+    }
+    return parentWindow;
+}
+
 void HideConsole(const v8::FunctionCallbackInfo<v8::Value>& args)
 {
-    ShowWindow(GetConsoleWindow(), SW_HIDE);
+    ShowWindow(GetTargetWindow(), SW_HIDE);
+    DWORD processId;
```
Comment (hetrodoo) [inline node-hide-console-window.cc]: Variable `processId` not referenced anywhere else.

**Review by hetrodoo — CHANGES_REQUESTED** [review]

File: node-hide-console-window.cc (hunk 1 above)
Comment (titushm) [reply node-hide-console-window.cc]: You are correct, I believe those lines were left in from a previous attempt to find a workaround

**Review by hetrodoo — APPROVED** [review]

**Comment by hetrodoo** [comment]
Deployed in version [2.2.0](https://www.npmjs.com/package/node-hide-console-window)

**Comment by FoxtrotSierra6829** [comment]
Thank you for fixing my issue! Working as intended now :)

**Pull request merged**